jwst/rscd/rscd_sub.py

```diff
@@ -1,15 +1,124 @@
+        science data to be corrected
+
+    rscd_model: rscd model object
+        rscd reference data
+
+    type: string
+        type of algorithm ['baseline' or 'enhanced']
+
+    Returns
+    -------
+    output_model: data model object
+        RSCD-corrected science data
+
+    """
+
+    output = input_model.copy()
+
+    # Retrieve the reference parameters for this exposure type
+    param = {}
+    param = get_rscd_parameters(input_model, rscd_model)
+
+    if param is None:
+        log.warning('RSCD correction will be skipped')
+        output.meta.cal_step.rscd = 'SKIPPED'
+        return output
+
+    if type == 'baseline':
+        group_skip = param['skip']
+        output = correction_skip_groups(input_model, group_skip)
+    else:
+        # enhanced algorithm is not enabled yet (updated code and validation needed)
+        log.warning('Enhanced algorithm not support yet: RSCD correction will be skipped')
+        output.meta.cal_step.rscd = 'SKIPPED'
+        return output
+        # decay function algorithm update needed
+        #output = correction_decay_function(input_model, param)
+
+    return output
+
+
+def correction_skip_groups(input_model, group_skip):
+    """
+    Short Summary
+    -------------
+    Set initial groups in integration to DO_NOT_USE to skip groups
+    affected by RSCD effect
+
+    Parameters
+    ----------
+    input_model: data model object
+        science data to be corrected
+
+    group_skip: int
+        number of groups to skip at the beginnig of the ramp
+
+    Returns
+    -------
+    output_model: data model object
+        RSCD-corrected science data
+    """
+
+    # Save some data params for easy use later
+    sci_nints = input_model.data.shape[0]       # number of integrations
+    sci_ngroups = input_model.data.shape[1]     # number of groups
+
+    log.debug("RSCD correction using: nints=%d, ngroups=%d" %
+              (sci_nints, sci_ngroups))
+
+    # Create output as a copy of the input science data model
+    output = input_model.copy()
+
+    if sci_ngroups <= (group_skip + 3):
+        log.warning("Too few groups to apply RSCD correction")
+        log.warning("RSCD step will be skipped")
+        output.meta.cal_step.rscd = 'SKIPPED'
+        return output
+
+    # Update the step status, and if ngroups > nflag+3, set all of the GROUPDQ in
+    # the first group to 'DO_NOT_USE'
+    # the +3 is to ensure there is a slope to be fit including the flagging for
+    # the last frame correction
+
+    # loop over all integrations except the first
+    mdelta = int(sci_nints / 10) + 1
+    for i in range(1, sci_nints):
+        if ((i + 1) % mdelta) == 0:
+            log.info(' Working on integration %d', i + 1)
+
+        output.groupdq[i, 0:group_skip :, :] = \
+            np.bitwise_or(output.groupdq[i, 0:group_skip,:,:], dqflags.group['DO_NOT_USE'])
+
+        log.debug(f"RSCD Sub: resetting GROUPDQ in the first {group_skip} groups to DO_NOT_USE")
```

Comment:
Minor nit, but you're not really resetting the GROUPDQ values, because it's a bitwise-OR operation, hence you're just adding DO_NOT_USE to any other flags that are already set.

Reply:
Fixed.